test/integration/controlplane/transformation/secrets_transformation_test.go

```diff
@@ -99,6 +120,513 @@ func TestSecretsShouldBeTransformed(t *testing.T) {
+			}
+
+			// d) re-create the secret
+			test.secret, err = test.createSecret(secretCorrupt, testNamespace)
+			if err != nil {
+				t.Fatalf("Failed to create test secret, error: %v", err)
+			}
+
+			// e) update the secret with a finalizer
+			withFinalizer := test.secret.DeepCopy()
+			withFinalizer.Finalizers = append(withFinalizer.Finalizers, "tes.k8s.io/fake")
+			test.secret, err = test.restClient.CoreV1().Secrets(testNamespace).Update(context.Background(), withFinalizer, metav1.UpdateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to add finalizer to the secret, error: %v", err)
+			}
+
+			test.runResource(test.TContext, unSealWithGCMTransformer, aesGCMPrefix, "", "v1", "secrets", test.secret.Name, test.secret.Namespace)
+
+			factory := informers.NewSharedInformerFactoryWithOptions(adminClient, time.Minute, informers.WithNamespace(testNamespace))
+			informer := factory.Core().V1().Secrets()
+
+			// the test adds a secret named "final-secret" after the unsafe delete
+			// so we can use it as an marker that the informer has caught up
+			finalEvent := make(chan struct{})
+			finalSecretName := "final-secret"
+			_, err = informer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+				AddFunc: func(obj interface{}) {
+					if obj, err := meta.Accessor(obj); err == nil && obj.GetName() == finalSecretName {
+						close(finalEvent)
+					}
+				},
+			})
+			if err != nil {
+				t.Fatalf("unexpected error from AddEventHandler: %v", err)
+			}
+
+			lister := informer.Lister()
+			factory.Start(test.Done())
+			waitForSyncCtx, waitForSyncCancel := context.WithTimeout(context.Background(), wait.ForeverTestTimeout)
+			defer waitForSyncCancel()
+			if !cache.WaitForCacheSync(waitForSyncCtx.Done(), informer.Informer().HasSynced) {
+				t.Fatalf("caches failed to sync")
+			}
+
+			// g) the secret should be readable from the cache
+			if _, err := lister.Secrets(testNamespace).Get(secretCorrupt); err != nil {
+				t.Errorf("unexpected failure getting the secret from lister cache: %v", err)
+			}
+
+			// h) override the config and break decryption of the old resources,
+			// the secret created in step b will be undecryptable
+			now := time.Now()
+			encryptionConf := filepath.Join(test.configDir, encryptionConfigFileName)
+			body, _ := ioutil.ReadFile(encryptionConf)
+			t.Logf("file before write: %s", body)
+			// we replace the existing key with a new key from a different provider
+			if err := os.WriteFile(encryptionConf, []byte(aesCBCConfigYAML), 0o644); err != nil {
+				t.Fatalf("failed to write encryption config that's going to make decryption fail")
+			}
+			body, _ = ioutil.ReadFile(encryptionConf)
+			t.Logf("file after write: %s", body)
+
+			// i) wait for the breaking changes to take effect
+			testCtx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			err = wait.PollUntilContextTimeout(testCtx, 1*time.Second, 2*time.Minute, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
```suggestion
			err = wait.PollUntilContextTimeout(testCtx, 1*time.Second, wait.ForeverTestTimeout, true, func(ctx context.Context) (done bool, err error) {
```

Reply:
unfortuately dynamic encryption config reload takes about 1m, so can't use wait.ForeverTestTimeout just yet, I have left to TODO to investigate and improve the reload time
```

Reply:
Add it to beta criteria or we will lose it.